### ISSUE TITLE
Result is both a namespace and a type

### ISSUE THREAD

**Issue by MisinformedDNA** [ardalis/Result#93]
As of #89, `Result` is both a namespace and a type. This is bad practice. Conceptually, the type name is ideal, but the namespace already exists.

Thoughts?

**Comment by ardalis** [comment]
Agreed and it's a problem. Obviously would be a breaking change. I think the type name should remain which means the namespace and possibly the package would need to change. Thoughts?

Ardalis.Result (obsolete/redundant)
Ardalis.ResultAbstraction
Ardalis (nothing else - would get confusing with all the other Ardalis.* packages, etc.)
Ardalis.Types
Ardalis.ReturnTypes
Ardalis.Abstractions

other ideas?

**Comment by MisinformedDNA** [comment]
I like `Ardalis` as the namespace and `Ardalis.Result` as the package name since it doesn't actually need it's own namespace. That's also true for most/all of your other libraries, most/all of them don't need their own namespace. If you don't want them in the base namespace, `Ardalis.Extensions` is a good option as well.

Giving a namespace to something as narrow as `Result` classes is hard since it's not a logical grouping of anything but itself.

But if you want to a separate namespace for just `Result` classes:

Ardalis.ResultAbstraction
Ardalis.ResultHelper
Ardalis.Results
Ardalis.OperationResult
Ardalis.SmartResults
Ardalis.ReturnResult

**Comment by ardalis** [comment]
Ardalis.Results came up in Discord and is one I'm leaning toward. Still considering if we should also do something to help differentiate from the new `IResult` type in minimal APIs in ASP.NET Core 6+, as well.

**Comment by KyleMcMaster** [comment]
Here are some example from two other popular result libraries.

This uses the root namespace `CSharpFunctionalExtensions` which would translate to simply `Ardalis`.
https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/Result/Result.cs

This uses `LanguageExt.Common`
https://github.com/louthy/language-ext/blob/main/LanguageExt.Core/Common/Result/Result.cs

Neither seem to be ideal for this situation so my vote would be to use the plural form `Ardalis.Results`.

**Comment by MisinformedDNA** [comment]
The namespace of the derived classes of `IResult` is `Microsoft.AspNetCore.Http.HttpResults`, which isn't too far off from `Results`.
